usd_qtpy/render_util/playblast.py

```diff
@@ -0,0 +1,215 @@
+
+    if TraverseAll:
+        gen = stage.TraverseAll()
+    else: 
+        gen = stage.Traverse()
+    
+    for prim in gen:
+        if prim.IsA(UsdGeom.Camera):
+            yield prim
+
+def camera_from_stageview(stage: Usd.Stage, stageview: Union[StageView, CustomStageView], name: str = "playblastCam") -> UsdGeom.Camera:
+    """ Catches a stage view whether it'd be from the custom viewer or from the baseclass and calls the export to stage function."""
+    stageview.ExportFreeCameraToStage(stage,name)
+    return UsdGeom.Camera.Get(stage,Sdf.Path(f"/{name}"))
+
+# Source: UsdAppUtils.colorArgs.py
+def get_color_args():
+    return ("disabled","sRGB","openColorIO")
+
+def get_complexity_levels() -> Generator[str]:
+    """
+    Returns a generator that iterates through all registered complexity presets in UsdAppUtils.complexityArgs
+    """
+    from pxr.UsdAppUtils.complexityArgs import RefinementComplexities as Complex
+    return (item.name for item in Complex._ordered)
+
+def iter_renderplugin_names() -> Generator[str]:
+    """
+    Returns a generator that will iterate through all names of Render Engine Plugin / Hydra Delegates
+    """
+    from pxr.UsdImagingGL import Engine as En
+    return (En.GetRendererDisplayName(pluginId) for pluginId in En.GetRendererPlugins())
+
+def get_renderplugin(enginestr: str):
+    from pxr.UsdImagingGL import Engine as En
+    for plug in En.GetRendererPlugins():
+        if enginestr == En.GetRendererDisplayName(plug):
+            return plug
+    return None
+
+def check_renderplugin_name(enginestr: str) -> Union[str, None]:
+    plugnames = iter_renderplugin_names()
+    if enginestr in plugnames:
+        return enginestr
+    return None
+
+def get_frames_string(start_time: int, end_time: int = None, frame_stride: float = None) -> str:
+    """
+    Takes a set of numbers and structures it so that it can be passed as frame string argument to e.g. render_playblast
+    Given only a start time, it'll render a frame at that frame.
+    Given a start and end time, it'll render a range from start to end, including end. (0-100 = 101 frames)
+    Given a start, end, and stride argument, it'll render a range with a different frame interval. 
+    (rendering every other frame can be done by setting this to 2.)
+    Output for 1, 2 and 3 arguments respectively: 
+    'start_time', 'start_time:end_time', 'start_time:end_timexframe_stride'
+    as defined by the USD standard.
+    """
+    # Keep adhering to USD standard as internally defined.
+    from pxr.UsdUtils import TimeCodeRange
+    range_token = TimeCodeRange.Tokens.RangeSeparator if end_time is not None else ""        # ":"
+    stride_token = TimeCodeRange.Tokens.StrideSeparator if frame_stride is not None else ""  # "x"
+    
+    end_time, frame_stride = end_time if end_time is not None else "", frame_stride if frame_stride is not None else ""
+
+    return f"{start_time}{range_token}{end_time}{stride_token}{frame_stride}"
```

Comment:
I wonder if this wouldn't be more readable:
```suggestion
    # Keep adhering to USD standard as internally defined.
    from pxr.UsdUtils import TimeCodeRange

    parts = [start_time]
    if end_time is not None:
        parts.append(TimeCodeRange.Tokens.RangeSeparator)  # ":"
        parts.append(end_time)
        
        if frame_stride is not None:
            parts.append(TimeCodeRange.Tokens.StrideSeparator)  # "x"
            parts.append(frame_stride)
            
    return "".join(parts)
```

Also, frame stride would mean nothing if no `end_time` is provided, right? So it should be nested within like this suggestion?

Reply:
yes, I had it branched like this before but got obsessed trying to reduce the amount of lines.
I do like aliasing the tokens because the namespaces are so ridiculously long.
Next push refactored as:

```py
from pxr.UsdUtils import TimeCodeRange
    range_token = TimeCodeRange.Tokens.RangeSeparator   # ":"
    stride_token = TimeCodeRange.Tokens.StrideSeparator  # "x"
    
    collect = f"{start_time}"

    if end_time is not None:
        collect += f"{range_token}{end_time}"
        if frame_stride is not None:
            collect += f"{stride_token}{frame_stride}"
    
    return collect
```

Reply:
Feel free to "mark resolved" the conversations here which you feel you have fixed in the code already.

Reply:
> Feel free to "mark resolved" the conversations here which you feel you have fixed in the code already.

I can't unfortunatly do that myself from here! I lack the permissions to do so, it seems.

Reply:
Thanks - I didn't know. :)